src/api/app/models/project.rb

```diff
@@ -963,12 +971,34 @@ def branch_to_repositories_from(project, pkg_to_enable, opts = {})
 
     return unless pkg_to_enable.is_channel?
 
-    # explizit call for a channel package, so create the repos for it
+    # explicit call for a channel package, so create the repos for it
     pkg_to_enable.channels.each do |channel|
       channel.add_channel_repos_to_project(pkg_to_enable)
     end
   end
 
+  def branch_remote_repositories(project_name, pkg_to_enable)
+    # handle remote package case
+    possible_ancestors = RemoteProject.new(name: project_name).possible_ancestor_names
+    possible_ancestors = [project_name] if possible_ancestors.size == 0
+
+    interconnect_project = RemoteProject.find_by(name: possible_ancestors)
+
+    return false unless interconnect_project
```

Comment:
Ok, not nice but fine! Please drop the false / true then as you don't check it anyway!